utils/install-app.js

```diff
@@ -139,6 +141,107 @@ function addToServer(appDir, installDir) {
+      recognizers[key].pluginVersion = appVers;
+      recognizers[key].pluginIdentifier = appId;
+      recognizers[key].key = appId + ":" + key + ":" + recognizers[key].id; // pluginid_that_provided_it:index(or_name)_in_that_provider:actionid
+    }
+    logger.debug('ZWED0294I', appId); //logger.debug("Found recognizers for '" + appId + "'");
+  } catch (e) {
+    logger.debug('ZWED0295I', (path.join(appDir, "config/recognizers", appId)), appId); //logger.debug("Could not find recognizers in '" + (path.join(appDir, "config/recognizers", appId)) + "'");
+  }
+
+  if (process.env.INSTANCE_DIR) {
+    configLocation = path.join(process.env.INSTANCE_DIR, "workspace/app-server/ZLUX/pluginStorage/org.zowe.zlux.ng2desktop/");
+  } else {
+    configLocation = path.join(userInput.zluxConfig, "../../ZLUX/pluginStorage/org.zowe.zlux.ng2desktop/");
+  }
+
+  try { // Get pre-existing recognizers in config, if any
+    configRecognizers = JSON.parse(fs.readFileSync(path.join(configLocation, "recognizers", appId))).recognizers;
+    const configRecognizersKeys = Object.keys(configRecognizers);
+    for (const configKey of configRecognizersKeys) { // Traverse config recognizers
+      for (const key of recognizerKeys) { // Traverse plugin recognizers
+        if (configRecognizers[configKey].key && recognizers[key].key && configRecognizers[configKey].key == recognizers[key].key) { // TODO: Need to implement real keys for Recognizers
+          configRecognizers[configKey] = recognizers[key]; // Choose the recognizers originating from plugin
+        }
+      }
+    }
+    recognizers = Object.assign(configRecognizers, recognizers); // // If found, combine the ones found in config with ones found in plugin
+    logger.debug('ZWED0296I', appId); //logger.debug("Found recognizers in config for '" + appId + "'");
+  } catch (e) {
+    logger.debug('ZWED0297I', appId); //logger.debug("No existing recognizers were found in config for '" + appId + "'");
+  }
+
+  if (recognizers) { // Attempt to copy recognizers over to config location for Desktop access later
+    try { //TODO: Doing recognizers.recognizers is redundant. We may want to consider refactoring in the future
+      fs.writeFileSync(path.join(configLocation, "recognizers", appId), '{ "recognizers":' + JSON.stringify(recognizers) + '}');
+      logger.debug('ZWED0298I', recognizers.length, appId); //logger.info("Successfully loaded " + recognizers.length + " recognizers for '" + appId + "' into config");
+    } catch (e) {
+      logger.debug('ZWED0299I', appId); //logger.debug("Unable to load recognizers for '" + appId + "' into config");
+    }
+  }
+}
+
+function copyActions(appDir, appId, appVers) {
+  let actions;
+  let actionsKeys;
+  let configActions;
+  let configLocation;
+
+  try { // Get actions in a plugin's appDir/config/xxx location
+    actions = JSON.parse(fs.readFileSync(path.join(appDir, "config/actions", appId))).actions;
```

Comment:
on the other hand this is perfectly fine. the app should only contain actions for itself, I think.